src/App.tsx

```diff
@@ -1,24 +1,244 @@
+
+          return newTodos;
+        });
+      });
+  };
+
+  const toggleAll = () => {
+    const allCompleted = uncompletedTodos.length === 0;
+    const todosToToggle = allCompleted ? completedTodos : uncompletedTodos;
+
+    todosToToggle.forEach(todo => {
+      handleTodoUpdate({ ...todo, completed: !allCompleted }).catch(() => {});
+    });
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this buttons is active only if there are some active todos */}
+          {!!todos.length && (
+            <button
+              type="button"
+              className={
+                classNames('todoapp__toggle-all',
+                  !!todos.every(t => t.completed) && 'active')
+              }
+              data-cy="ToggleAllButton"
+              onClick={toggleAll}
+            />
+          )}
+
+          {/* Add a todo on form submit */}
+          <form onSubmit={handleSubmit}>
+            <input
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              disabled={isDisabled}
+              ref={titleField}
+              value={todoTitle}
+              onChange={handleTitleChange}
+            />
+          </form>
+        </header>
+
+        {!!todos.length && (
+          <>
+            <TodoList
+              todos={filteredTodos}
+              deleteTodo={deleteTodo}
+              tempTodo={tempTodo}
+              onUpdateTodo={handleTodoUpdate}
+            />
+            <footer className="todoapp__footer" data-cy="Footer">
+              <span className="todo-count" data-cy="TodosCounter">
+                {`${uncompletedTodos.length} items left`}
+              </span>
+
+              <TodoFilter changeFilter={setFilterBy} />
+
+              <button
+                type="button"
+                className="todoapp__clear-completed"
+                data-cy="ClearCompletedButton"
+                onClick={onClearCompleted}
+                disabled={completedTodos.length === 0}
+              >
+                Clear completed
+              </button>
+
+            </footer>
+          </>
+        )}
+      </div>
+      <div
+        data-cy="ErrorNotification"
+        className={
+          classNames('notification is-danger is-light has-text-weight-normal',
+            !errorMasage && 'hidden')
```

Comment:
You are not currently using this library to conditionally add classes, this needs to be fixed everywhere